sql/hive/src/main/scala/org/apache/spark/sql/hive/SQLBuilder.scala

```diff
@@ -233,6 +242,67 @@ class SQLBuilder(logicalPlan: LogicalPlan, sqlContext: SQLContext) extends Loggi
     )
   }
 
+  private def groupingSetToSQL(
+      plan: Aggregate,
+      expand: Expand,
+      project: Project): String = {
+    require(plan.groupingExpressions.length > 1)
+
+    // The last column of Expand is always grouping ID
+    val gid = expand.output.last
+
+    val groupByAttributes = plan.groupingExpressions.dropRight(1).map(_.asInstanceOf[Attribute])
+    val groupByExprs =
+      project.projectList.drop(project.child.output.length).map(_.asInstanceOf[Alias].child)
```

Comment:
nit: we can define `val numOriginalOutput = project.child.output.length` first, and make `val groupByExprs = ...` fit it one line. Also add comment to explain the assumption here please.